### DISCUSSION TITLE
Google Analytics Import - 500 http response

### DISCUSSION THREAD

**Discussion by jacobawebb** [plausible/analytics#3080] [Self-Hosted Support]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Using official Plausible Cloud hosting or self-hosting?

Self-hosting

### Describe the bug

I currently have Plausible running on a private VPS behind an NGINX proxy, everything is working as expected apart from the Google Analytics import process.

Trying to run it I get a 500 error. It's not a Google API issue, I already ran an import to Matomo with no issues apart from rate limiting.

Everything should work and it doesn't throw any "Google Client Secret missing" sort of errors.

Docker console:
```console
hosting-plausible-1            | Request: GET /auth/google/callback?state=%5B1,%22import%22%5D&code={CODE_HERE}&scope=email%20https://www.googleapis.com/auth/analytics.readonly%20openid%20https://www.googleapis.com/auth/userinfo.email&authuser=0&prompt=consent
hosting-plausible-1            | ** (exit) an exception was raised:
hosting-plausible-1            |     ** (MatchError) no match of right hand side value: {:error, %Plausible.HTTPClient.Non200Error{reason: %Finch.Response{body: %{"error" => "invalid_client", "error_description" => "Unauthorized"}, headers: [{"cache-control", "no-cache, no-store, max-age=0, must-revalidate"}, {"date", "Mon, 26 Jun 2023 11:42:48 GMT"}, {"expires", "Mon, 01 Jan 1990 00:00:00 GMT"}, {"pragma", "no-cache"}, {"content-type", "application/json; charset=utf-8"}, {"vary", "X-Origin"}, {"vary", "Referer"}, {"server", "scaffolding on HTTPServer2"}, {"x-xss-protection", "0"}, {"x-frame-options", "SAMEORIGIN"}, {"x-content-type-options", "nosniff"}, {"alt-svc", "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"}, {"accept-ranges", "none"}, {"vary", "Origin,Accept-Encoding"}, {"transfer-encoding", "chunked"}], status: 401}}}
hosting-plausible-1            |         (plausible 0.0.1) lib/plausible/google/http.ex:128: Plausible.Google.HTTP.fetch_access_token/1
hosting-plausible-1            |         (plausible 0.0.1) lib/plausible_web/controllers/auth_controller.ex:565: PlausibleWeb.AuthController.google_auth_callback/2
hosting-plausible-1            |         (plausible 0.0.1) lib/plausible_web/controllers/auth_controller.ex:1: PlausibleWeb.AuthController.action/2
hosting-plausible-1            |         (plausible 0.0.1) lib/plausible_web/controllers/auth_controller.ex:1: PlausibleWeb.AuthController.phoenix_controller_pipeline/2
hosting-plausible-1            |         (phoenix 1.6.6) lib/phoenix/router.ex:355: Phoenix.Router.__call__/2
hosting-plausible-1            |         (plausible 0.0.1) lib/plausible_web/endpoint.ex:1: PlausibleWeb.Endpoint.plug_builder_call/2
hosting-plausible-1            |         (plausible 0.0.1) lib/plausible_web/endpoint.ex:1: PlausibleWeb.Endpoint."call (overridable 3)"/2
hosting-plausible-1            |         (plausible 0.0.1) lib/plausible_web/endpoint.ex:1: PlausibleWeb.Endpoint.call/2
```

Client console (browser):
![image](https://github.com/plausible/analytics/assets/95496857/7e593b42-0c36-4752-b684-99207bee9516)

I have followed all the relevant documentation:
- https://plausible.io/docs/google-analytics-import
- https://plausible.io/docs/self-hosting-configuration#google-api-integration
- https://plausible.io/docs/self-hosting-configuration#google-search-integration

`plausible-conf.env`
![image](https://github.com/plausible/analytics/assets/95496857/bd251941-a89b-4d1b-a598-912a8a85d313)

Journey:

1.
![image](https://github.com/plausible/analytics/assets/95496857/08947413-b8af-404d-ae41-0a200622e638)

2.
![image](https://github.com/plausible/analytics/assets/95496857/42bc1f4c-43e2-40ab-91d5-e154f79af2d8)

3.
![image](https://github.com/plausible/analytics/assets/95496857/ef1e11e3-7324-426d-a3a4-2553fbf22963)

4.
![image](https://github.com/plausible/analytics/assets/95496857/e15e9e68-6483-40a8-ba21-dc5e394e5894)

5.
![image](https://github.com/plausible/analytics/assets/95496857/b78f049e-c50b-4fc5-9e7e-9cec5bf598a4)

### Expected behavior

Google Analytics Data to begin importing with no issues.

### Screenshots

See bug description

### Environment

```markdown
- OS: Windows 11 Client / AlmaLinux 8.8 Server
- Browser: Chrome
- Browser Version: 114.0.5735.134 (Official Build) (32-bit)
```

**Comment by clementmas** [comment]
I think you missed the first step of the guide:

> [Google Analytics Integration](https://plausible.io/docs/self-hosting-configuration#google-api-integration)
> To be able to import data from Google Analytics, **you need to enable [Analytics Reporting API](https://console.developers.google.com/apis/api/analyticsreporting.googleapis.com) and [Google Analytics API](https://console.developers.google.com/apis/api/analytics.googleapis.com)**.
> 

I had the same issue but then I fixed it by enabling the "Analytics Reporting API".

@Plausible: that'd be good to see the error message on the 500 page.
